File: javascript/dp/198_House_Robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    let rob1= 0, rob2 = 0

    for (n of nums) {
        temp = Math.max(rob1 + n, rob2);
        rob1 = rob2;
        rob2 = temp;
    }
    return rob2;
};

File: python/dp/198_House_Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        rob1, rob2 = 0, 0 

        # [rob1, rob2, n, n+1, ...]
        for n in nums:
            temp = max(rob1 + n, rob2)  # new max
            rob1 = rob2  # previous max
            rob2 = temp

        return rob2  